src/stage1/bigint.cpp

```diff
@@ -468,6 +468,76 @@ void bigint_min(BigInt* dest, const BigInt *op1, const BigInt *op2) {
     }
 }
 
+/// clamps op within bit_count/signedness boundaries
+void bigint_clamp_by_bitcount(BigInt* dest, uint32_t bit_count, bool is_signed) {
+    bool is_negative = dest->is_negative;
+    // set to false so that bigint_bits_needed() won't report 64+ bits needed for small negative numbers
```

Comment:
Elaborating that is_negative is a workaround of the biting_bits_needed implementation returning 0 for negative numbers in comments would be helpful.
This only explains the outcome.

Reply:
Thanks! I can see how the comments/code may not have been easy to read.  Hope the most recent commits help with that.  Added those test cases too. :+1:

Reply:
I marked the conversation as resolved, but do lLet me know if i missed anything.